src/sentry/incidents/logic.py

```diff
@@ -629,6 +631,13 @@ def create_alert_rule(
             detection_type=detection_type,
         )
 
+        if alert_rule.detection_type == AlertRuleDetectionType.DYNAMIC.value:
+            if not features.has("organizations:anomaly-detection-alerts", organization, actor=user):
+                alert_rule.delete()
+                raise ResourceDoesNotExist
```

Comment:
Can we add details to this exception?

Maybe just a `Feature is not available`?